Development instructions + link from README + additional make targets

brandur opened the pull request:
A few miscellaneous changes that are largely documentation related:

* Add `docs/development.md` containing basic instructions on how to
  install dependencies, run tests, run lint, run code formatting, etc.
  This matches convention in other River projects and makes it easier
  for people to bootstrap themselves when they're not deeply familiar
  with Python's conventions already.

* Link `docs/development.md` from top-level README. This pull originally
  proposed also moving the README to `docs/` for consistency with other
  River projects, but apparently Rye does not allow this.

* Add some additional make targets for other development tasks beyond
  the type check. Again, not strictly necessary, but sort of nice
  because it means you don't have to remember which targets need to use
  make versus which need to use Rye -- you can just `make` anything and
  it'll work. Also, make all targets Phony [1].

[1] https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

brandur committed:
Development instructions + link from README + additional make targets

A few miscellaneous changes that are largely documentation related:

* Add `docs/development.md` containing basic instructions on how to
  install dependencies, run tests, run lint, run code formatting, etc.
  This matches convention in other River projects and makes it easier
  for people to bootstrap themselves when they're not deeply familiar
  with Python's conventions already.

* Link `docs/development.md` from top-level README. This pull originally
  proposed also moving the README to `docs/` for consistency with other
  River projects, but apparently Rye does not allow this.

* Add some additional make targets for other development tasks beyond
  the type check. Again, not strictly necessary, but sort of nice
  because it means you don't have to remember which targets need to use
  make versus which need to use Rye -- you can just `make` anything and
  it'll work. Also, make all targets Phony [1].

[1] https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

brandur commented on docs/README.md:
```diff
@@ -0,0 +1,7 @@
+
+An insert-only Python client for [River](https://github.com/riverqueue/river) packaged in the [`riverqueue` gem](https://rubygems.org/gems/riverqueue). Allows jobs to be inserted in Python and run by a Go worker, but doesn't support working jobs in Python.
+
+## Development
```
README still needs more work on usage, but skipping that for now.

brandur commented on docs/README.md:
```diff
@@ -0,0 +1,7 @@
+# River client for Python
+
+An insert-only Python client for [River](https://github.com/riverqueue/river) packaged in the [`riverqueue` gem](https://rubygems.org/gems/riverqueue). Allows jobs to be inserted in Python and run by a Go worker, but doesn't support working jobs in Python.
```
Changed this just so the copy matches riverqueue-ruby exactly for consistency.

brandur commented:
@ewhauser Mind taking a review pass on this one?

ewhauser commented:
@brandur `rye` requires a `README.md` at the root of the repo which is why the build is failing. Can just be an empty file.

brandur commented:
@ewhauser Ah thank you — okay I changed the README back and updated the commit message. OOC, do you know what Rye is doing with the README? I did a quick Google and couldn't find much info on this.

ewhauser commented:
Not 100% why it is necessary either, but `rye` uses `hatchling` under the covers where is where the error was coming from.

brandur commented:
Ah k. Okay cool. Thanks!
